collated/main/A0141128R.md

```diff
@@ -0,0 +1,692 @@
+        	}
+    		return new BlockCommand(
+    			detailToAdd,
+    			dtStart.toLocalDate(),
+    			dtStart.toLocalTime(),
+    			dtEnd.toLocalDate(),
+    			dtEnd.toLocalTime(),
+    			"",
+    			getTagsFromArgs(splittedArgs));
+    	} catch (IllegalValueException ive) {
+    		return new IncorrectCommand(ive.getMessage());
+    	}
+    }
+    
+    /**
+     * Parses arguments in the context of the edit task command.
+     * 
+     * @param args full command args string
+     * @return the prepared command
+     * 
+```
+###### /java/seedu/cmdo/logic/parser/MainParser.java
+``` java
+     */
+    private Command prepareEdit(String args){
+    	try {
+    	// Determine if edit command is input correctly
+    	Optional<Integer> checkForIndex = parseLooseIndex(args);
+    	
+        if(!checkForIndex.isPresent()){
+            return new IncorrectCommand(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE));
+        }
+    	//check for empty detail
+        if (args.lastIndexOf("'") == args.indexOf("'"))
+        	detailToAdd = "";
+        // Determine if the edit command is used correctly
+    	String[] splittedArgs = getCleanString(args).split(" ");
+        
+    	Integer index = Integer.valueOf(splittedArgs[0]);
+        if(index == null){
+            return new IncorrectCommand(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE));
+        }
+        // Store index and remove
+        int targetIndex = index;
+        args = args.replaceFirst("[0-9]+\\s", "");
+        //If details is not empty, extract details
+        if(detailToAdd == null || !detailToAdd.equals(""))
+        extractDetail(args);
+        //used a flag to check if floating task
+        boolean floating = false;
+        //used flag to check if want to remove priority
+        boolean removePriority = false;
+        // Parse date and time
+        reducedArgs = extractDueByDateAndTime(args);
+        //if keyword float is entered, it becomes a floating task (no date no time)
+        if(reducedArgs.toLowerCase().contains("floating")){
+        	floating = true;
+        }
+        //if keyword rp or remove priority is entered, priority is removed
+        if(reducedArgs.toLowerCase().contains("remove priority")||reducedArgs.toLowerCase().contains("rp")){
+        	removePriority = true;
+        }
+        // used as flag for task type. 0 for floating, 1 for non-range, 2 for range
+    	int dataNo;
+        LocalDateTime dt = LocalDateTime.MIN;
+    	LocalDateTime dtStart = LocalDateTime.MIN;
+    	LocalDateTime dtEnd = LocalDateTime.MIN;
+    	
+    	if (datesAndTimes.size() == 1) {
+    		dt = datesAndTimes.get(0);
+    		dataNo = 1;
+    	} else if (datesAndTimes.size() == 2) {
+    		dtStart = datesAndTimes.get(0);
+    		dtEnd = datesAndTimes.get(1);
+    		dataNo = 2;
+    	} else{
+    		dt = LocalDateTime.of(NO_DATE_DEFAULT, LocalTime.MAX);
+        			dataNo = 0;
```

Comment:
\O/ this is an extremely long code, usualy a hint for SLAP ~ try to improve SLAP here